storage/statedb/database.go

```diff
@@ -605,26 +596,17 @@ func (db *Database) DoesExistNodeInPersistent(hash common.Hash) bool {
 
 // preimage retrieves a cached trie node pre-image from memory. If it cannot be
 // found cached, the method queries the persistent database for the content.
-func (db *Database) preimage(hash common.Hash) ([]byte, error) {
+func (db *Database) preimage(hash common.Hash) []byte {
 	// Retrieve the node from cache if available
 	db.lock.RLock()
 	preimage := db.preimages[hash]
 	db.lock.RUnlock()
 
 	if preimage != nil {
-		return preimage, nil
+		return preimage
 	}
 	// Content unavailable in memory, attempt to retrieve from disk
-	return db.diskDB.ReadCachedTrieNodePreimage(secureKey(hash))
-}
-
-// secureKey returns the database key for the preimage of key (as a newly
-// allocated byte-slice)
-func secureKey(hash common.Hash) []byte {
-	buf := make([]byte, secureKeyLength)
-	copy(buf, secureKeyPrefix)
-	copy(buf[secureKeyPrefixLength:], hash[:])
-	return buf
+	return db.diskDB.ReadPreimage(hash)
```

Comment:
is this okay to change key?

Reply:
Actually, the secureKeyPrefix is equal to preimagePrefix, which is "secure-key-".

Reply:
I got it. Thanks.